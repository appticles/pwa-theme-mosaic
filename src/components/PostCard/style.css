/* OVER RIDE semantic UI styles */
/* Image Element */
.item.posts div.image {
  background-size: cover;
  height: 40%;
  width: 50%;
  float: left;
  margin: 4px 10px 5px 0;
}

/* Item */
.posts {
  text-align: start;
}
.posts .header {
  height: 10%;
  margin: 10px 0;
  font-size: 1.6rem;
  font-weight: 100;
  line-height: 19px;
  color: #000000;
  text-align: start;
  text-transform: uppercase;
}

.ui.card a {
  height: 100%;
}
