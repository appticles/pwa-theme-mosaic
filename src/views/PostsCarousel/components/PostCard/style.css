/* OVER RIDE semantic UI styles */
/* Image Element */
.item.posts div.image {
  background-size: cover;
  background-position: center;
  height: 40%;
  width: 50%;
  float: left;
  margin: 4px 10px 5px 0;
}

/* Item */
.posts {
  text-align: start;
}
.posts .header {
  height: 8.2%;
  margin: 10px 0;
  font-size: 1.6rem;
  font-weight: 100;
  line-height: 23px;
  color: #000000;
  text-align: start;
  text-transform: uppercase;
}

.posts .header {
  overflow: hidden;
  text-overflow: ellipsis;
}

.ui.card a {
  height: 100%;
}

.item.posts .description {
  text-overflow: ellipsis;
  word-break: break-word;
}
