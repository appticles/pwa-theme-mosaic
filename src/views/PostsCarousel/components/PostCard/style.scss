@import "../../../../utilities/_variables.scss";

.item.post-item {
  padding: 1rem;
  .header {
    text-transform: uppercase;
    overflow: hidden;
    margin-top: $base-spacing-sm;
    margin-bottom: $base-spacing-sm;
  }

  .content {
    padding-top: $base-spacing-xsm;
    .image {
      background: center;
      background-size: cover;
      float: left;
      height: 120px;
      width: 50%;
      max-width: 200px;
      margin-top: $base-spacing-xs;
      margin-bottom: $base-spacing-xs;
      margin-right: $base-spacing-xsm;
      margin-left: $base-spacing-xxs;
    }
    .description {
      text-align: justify;
      word-break: break-word;
    }
  }
}
