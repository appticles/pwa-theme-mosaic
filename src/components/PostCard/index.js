import React, { Component } from 'react';
import { Item } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

import { postPropType } from '../../views/PostsCarousel/reducer';
import './style.css';

export default class PostCard extends Component {
  render() {
    const { post } = this.props;
    return (
      <Link to={'/post/' + post.slug + '/' + post.id}>
        <Item className="posts">
          <Item.Header>{post.title.rendered}</Item.Header>
          <Item.Content>
            <Item.Image size="large" src="http://via.placeholder.com/177x119" />
            <Item.Description dangerouslySetInnerHTML={{ __html: post.excerpt.rendered }} />
          </Item.Content>
        </Item>
      </Link>
    );
  }
}

PostCard.propTypes = {
  post: postPropType.isRequired,
};
