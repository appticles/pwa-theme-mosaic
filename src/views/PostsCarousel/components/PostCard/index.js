import React from 'react';
import { Item } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import _ from 'lodash';

import { postPropType } from '../../reducer';
import './style.css';

const PostCard = (props) => {
  function getImage(sourceImg) {
    if (_.isNil(sourceImg) && sourceImg !== 0) {
      return props.post._embedded['wp:featuredmedia'][0].source_url;
    }

    return null;
  }

  let path = {};

  if (_.isNil(props.category)) {
    path = '/post/' + props.post.slug + '/' + props.post.id;
  }
  path =
    '/category/' + props.category.params.categorySlug + '/' + props.category.params.categoryId + '/post/' + props.post.slug + '/' + props.post.id;

  return (
    <Link to={path}>
      <Item className="posts">
        {props.post.title && <Item.Header dangerouslySetInnerHTML={{ __html: props.post.title.rendered }} />}
        <Item.Content>
          <div
            className="image"
            style={{
              backgroundImage: 'url(' + getImage(props.post.featured_media) + ')',
            }}
          />
          <Item.Description dangerouslySetInnerHTML={{ __html: props.post.excerpt.rendered }} />
        </Item.Content>
      </Item>
    </Link>
  );
};

PostCard.propTypes = {
  post: postPropType.isRequired,
  category: PropTypes.shape({
    params: PropTypes.shape({
      categorySlug: PropTypes.string.isRequired,
      categoryId: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default PostCard;
