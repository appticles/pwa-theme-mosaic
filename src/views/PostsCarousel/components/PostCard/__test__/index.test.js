import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import PostCard from '../index';

describe('Post card', () => {
  it('should render correctly post with featured image', () => {
    const mockPost = {
      id: 1,
      title: {
        rendered: 'Mock title',
      },
      author: 55,
      categories: [43, 54, 23],
      slug: 'mock-slug',
      content: {
        rendered: 'Some mock content to be rendered',
        protected: false,
      },
      excerpt: {
        rendered: 'Some mock exceprt to be rendered',
        protected: true,
      },
      featured_media: 23,
      date: 'some mock date',
      _embedded: {
        'wp:featuredmedia': 'someMockImageUrl',
      },
    };
    const category = {
      params: {
        categoryId: '32',
        categorySlug: 'some-slug',
      },
    };
    const output = shallow(<PostCard post={mockPost} category={category} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render correctly post with placeholder image', () => {
    const mockPost = {
      id: 1,
      title: {
        rendered: 'Mock title',
      },
      author: 55,
      categories: [43, 54, 23],
      slug: 'mock-slug',
      content: {
        rendered: 'Some mock content to be rendered',
        protected: false,
      },
      excerpt: {
        rendered: 'Some mock exceprt to be rendered',
        protected: true,
      },
      featured_media: 0,
      date: 'some mock date',
    };
    const category = {
      params: {
        categoryId: '32',
        categorySlug: 'some-slug',
      },
    };
    const output = shallow(<PostCard post={mockPost} category={category} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
