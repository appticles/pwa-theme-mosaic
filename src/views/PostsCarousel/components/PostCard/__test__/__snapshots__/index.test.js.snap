// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post card should render correctly post with featured image 1`] = `
<Link
  replace={false}
  to="/category/some-slug/32/post/mock-slug/1"
>
  <Item
    className="post-item"
  >
    <ItemHeader
      as="h1"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Mock title",
        }
      }
    />
    <ItemContent>
      <div
        className="image"
        style={
          Object {
            "backgroundImage": "url(undefined)",
          }
        }
      />
      <ItemDescription
        dangerouslySetInnerHTML={
          Object {
            "__html": "Some mock exceprt to be rendered",
          }
        }
      />
    </ItemContent>
  </Item>
</Link>
`;

exports[`Post card should render correctly post with placeholder image 1`] = `
<Link
  replace={false}
  to="/category/some-slug/32/post/mock-slug/1"
>
  <Item
    className="post-item"
  >
    <ItemHeader
      as="h1"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Mock title",
        }
      }
    />
    <ItemContent>
      <ItemDescription
        dangerouslySetInnerHTML={
          Object {
            "__html": "Some mock exceprt to be rendered",
          }
        }
      />
    </ItemContent>
  </Item>
</Link>
`;
